metals/src/main/scala/scala/meta/internal/metals/testProvider/frameworks/JunitTestFinder.scala

```diff
@@ -37,7 +39,15 @@ class JunitTestFinder {
         case symbol if isValid(symbol) =>
           doc
             .toLocation(uri, symbol.symbol)
-            .map(location => TestCaseEntry(symbol.displayName, location))
+            .map { location =>
+              val encodedName = NameTransformer.encode(symbol.displayName)
+              TestCaseEntry(
```

Comment:
Could we instead encode the name in `DebugProvider.startTestSuite` ? 

Something like:
```
            request.requestData
              .copy(suites =
                request.requestData.suites.map(su =>
                  su.copy(tests =
                    su.tests.map(tc => NameTransformer.encode(tc).replace("$", "\\$"))
                  )
                )
              )
              .toJson,
```
but nicer?

Wouldn't that help with other testing frameworks? Or at least not break them? This way we wouldn't need another PR to VS Code extension

Reply:
I don't think it can be done in a way that doesn't need a PR in the extension, unless we are okay with `example-name` to be visible as `example$eqname` in the test explorer (for JUnit). 

In JUnit `example-name` has to be send as `example\$eqname` but `example$eqname` will be returned by debug adapter with the result of the test. So VS code needs to know to match on `example$eqname` to show correctly test results. So we don't need to send three different names but still we'd need a display name.

This PR shouldn't break anything since this never worked.

Reply:
Sure, I was just trying to figure out if we can fix things even if anyone implemented the current protocol.